Streaming uploads

xloem opened the pull request:
This is an extension on #36 to stream all files.

python-requests streams file objects and iterators if they are passed as post body content.  I've changed all single file uploads to upload in that way.

m-cat commented:
Merge conflict needs to be resolved

xloem commented:
Thanks.  Resolved.

m-cat commented on tests/test_integration_upload.py:
```diff
@@ -13,6 +13,23 @@
 client = skynet.SkynetClient()
 
 
+def response_callback(request):
```
I don't understand what this function does or why we started using it as a callback. Can you please add a docstring for the function and some comments?

xloem replied:
it's a way responses has for handling requests.  it could be used to do server actions like providing the appropriate skylink and response for the data.  i use it to simply gather the uploaded data so it can be verified the code under test uploaded the right data.

m-cat commented on tests/test_integration_upload.py:
```diff
@@ -36,16 +52,17 @@ def test_upload_file():
 
     headers = responses.calls[0].request.headers
-    assert headers["Content-Type"].startswith("multipart/form-data;")
+    assert headers["Content-Type"]
```
q: Why did this change?

xloem replied:
requests doesn't do multipart streaming upload, so single file uploads are no longer multipart

m-cat commented on tests/test_integration_upload.py:
```diff
@@ -36,16 +52,17 @@ def test_upload_file():
+    assert params["filename"] == "file1"
+
     body = responses.calls[0].request.body
-    assert str(body).find('Content-Disposition: form-data; name="file"; \
```
q: Why was this removed?

xloem replied:
same reason as above: the content data no longer has multipart decorations

m-cat reviewed (changes requested):
One requested change plus some questions.

xloem commented:
Sorry about all that, looks like I made an error rebasing.  I think only the first change is relevant and don't immediately recognise the others.  Consider a draft until addressed.

But if you want streaming to work basically you just pass a file object straight to a single file post just like with chunked uploading.

xloem commented:
Sorry for misleading comment.  The changes are to make the tests pass, after I changed single file uploads to be streaming.  I also added tests that the body content is correct.

I added the comments you requested.

m-cat reviewed (approved):
Looks great. Thanks for the comments 🙂
